Utilities/Cleanup-AzureStorageAccounts.ps1

```diff
@@ -0,0 +1,310 @@
+        -Patterns "" -CleanupAgeInDays 365 -DryRun
+
+Important Notes:
+================
+    1.  Empty <-Patterns ""> value selects all the files in given containers.
+    2.  If you are not sure about what is being cleaned up, then use -DryRun mode.
+        Then, examine the .csv file created, which shows list of files and cleanup status.
+        If you're satisfied with dryrun cleanup list, remove -DryRun and run the cleanup.
+#>
+
+param(
+    [String] $SecretFilePath,
+
+    # Comma separated storage account names. If empty, All storage accounts in current subscriptions will be selected.
+    [string] $StorageAccountNames,
+
+    # Comma separated skipped Storage account names.
+    [string] $SkippedStorageAccountNames,
+
+    # Comma separated Azure regions. If empty, All regions in current subscriptions will be selected.
+    [string] $AzureRegions,
+
+    # Comma separated values of exact VHD names to clean.
+    [String] $VHDNames,
+
+    # Command separated VHD name patterns to clean.
+    [String] $Patterns = "LISAv2,-osdisk.vhd,AUTOBUILT",
+
+    # Cleanup Age (in Days)
+    [int] $CleanupAgeInDays = 7,
+
+    # Switch to enable logs for skipped files. (Makes execution slower).
+    [switch] $ShowSkippedFiles,
+
+    # Switch to enable dry run mode.
+    [switch] $DryRun
+)
+
+
+try {
+    #region Authenticate the powershell session.
+    if ($SecretFilePath) {
+        .\Utilities\AddAzureRmAccountFromSecretsFile.ps1 -customSecretsFilePath $SecretFilePath
+        $SubID = ([xml](Get-Content $SecretFilePath)).Secrets.SubscriptionID
+    } else {
+        Write-LogInfo "Secret File not provided. Checking subscriptions..."
+        $Subscriptions = Get-AzSubscription
+        if ($Subscriptions) {
+            $SubscriptionCount = $Subscriptions.Count
+            if ($SubscriptionCount -eq 1) {
+                $SubID = $Subscriptions.Id
+                $SubName = $Subscriptions.Name
+            } else {
+                $ChoiceString = ""
+                $ChoiceCounter =  1
+                $Subscriptions | ForEach-Object {
+                    Write-LogInfo "$ChoiceCounter. $($_.Name) ($($_.Id))"
+                    $ChoiceString += "$ChoiceCounter/"
+                    $ChoiceCounter += 1;
+                }
+                $ChoiceString = $ChoiceString.TrimEnd('/')
+                $UserChoice = 0
+                while ($UserChoice -gt ( $ChoiceCounter -1 ) -or $UserChoice -le 0) {
+                    Write-LogWarn "Please make sure that you choose correct subscription for storage cleanup."
+                    $UserChoice = [int] (Read-Host "Enter your choice. [$ChoiceString]")
+                }
+                $SubID = $Subscriptions[$UserChoice -1].Id
+                $SubName = $Subscriptions[$UserChoice -1].Name
+            }
+            Write-LogInfo "Selected - '$SubName ($SubID)'."
+
+            # Get the final confirmation from user
```

Comment:
why do this in a script that is to run unattended?

Reply:
If user gives secret file, then script will run unattended, else it will require user confirmation.

Reply:
ok, if so, please add that instruction in the comment of Head part

Reply:
> ok, if so, please add that instruction in the comment of Head part

@iamshital any followup?